### PR TITLE
RR-939 - Client and service methods for Create Education, and associated code

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
This PR creates the client and service methods for Create Education, and associated code.

### CODE REVIEW

File: server/@types/educationAndWorkPlanApi/index.d.ts [hunk 1]
```diff
@@ -4,6 +4,54 @@
  */
 
 export interface paths {
+  '/queue-admin/retry-dlq/{dlqName}': {
```
Comment (nathanrussell-moj-digital) [inline server/@types/educationAndWorkPlanApi/index.d.ts:7]: I had to re-import the API swagger spec to get the data types for CreateEducationRequest etc.
But it also pulled in these changes to do with DLQ endpoints. These are not related to what we are doing in this PR, but it is not a problem. It just means we've not re-imported the spec for a while and there have been some unrelated changes happening in the API codebase recently.

File: server/@types/educationAndWorkPlanApi/index.d.ts [hunk 2]
```diff
@@ -760,6 +832,60 @@ export interface components {
        */
       reasonOther?: string
     }
+    /**
+     * @description A list of achieved qualifications that should be created as part of the education record.
+     * @example null
+     */
+    CreateAchievedQualificationRequest: {
```
Comment (nathanrussell-moj-digital) [inline server/@types/educationAndWorkPlanApi/index.d.ts:839]: These 2 data types are the 2 new ones we are looking to use in this PR 👍

File: server/data/educationAndWorkPlanClient.ts [hunk 3]
```diff
@@ -109,16 +110,27 @@ export default class EducationAndWorkPlanClient {
     prisonNumber: string,
     createInductionRequest: CreateInductionRequest,
     token: string,
-  ): Promise<never> {
+  ): Promise<void> {
```
Comment (nathanrussell-moj-digital) [inline server/data/educationAndWorkPlanClient.ts:113]: Minor unrelated tidying

File: server/data/educationAndWorkPlanClient.ts [hunk 4]
```diff
@@ -109,16 +110,27 @@ export default class EducationAndWorkPlanClient {
...
     return EducationAndWorkPlanClient.restClient(token).post({
       path: `/inductions/${prisonNumber}`,
       data: createInductionRequest,
     })
   }
 
-  async getEducationResponse(prisonNumber: string, token: string): Promise<EducationResponse> {
+  async getEducation(prisonNumber: string, token: string): Promise<EducationResponse> {
```
Comment (nathanrussell-moj-digital) [inline server/data/educationAndWorkPlanClient.ts:120]: Minor tidying of this method name, just to make it consistent with the other method names in the client classes

File: server/data/educationAndWorkPlanClient.ts [hunk 5]
```diff
@@ -109,16 +110,27 @@ export default class EducationAndWorkPlanClient {
...
     return EducationAndWorkPlanClient.restClient(token).get<EducationResponse>({
       path: `/person/${prisonNumber}/education`,
     })
   }
+
+  async createEducation(
```
Comment (nathanrussell-moj-digital) [inline server/data/educationAndWorkPlanClient.ts:126]: This is the new client method for Create Education

File: server/data/mappers/createOrUpdateInductionDtoMapper.test.ts [hunk 6]
```diff
@@ -1,13 +1,13 @@
 import toCreateOrUpdateInductionDto from './createOrUpdateInductionDtoMapper'
 import { aValidInductionDto } from '../../testsupport/inductionDtoTestDataBuilder'
-import aValidCreateOrUpdateInductionDto from '../../testsupport/updateInductionDtoTestDataBuilder'
+import aValidUpdateInductionDto from '../../testsupport/updateInductionDtoTestDataBuilder'
```
Comment (nathanrussell-moj-digital) [inline server/data/mappers/createOrUpdateInductionDtoMapper.test.ts:3]: Unrelated, but I took the opportunity to tidy the names of a couple of test data builder functions for `InductionDto`s, just to make them consistent with other test data builder functions 👍

File: server/services/educationAndWorkPlanService.ts [hunk 7]
```diff
@@ -66,22 +74,36 @@ export default class EducationAndWorkPlanService {
     return this.educationAndWorkPlanClient.unarchiveGoal(unarchiveGoalDto.prisonNumber, unarchiveGoalRequest, token)
   }
 
-  private async getAllPrisonNamesByIdSafely(token: string): Promise<Map<string, string>> {
```
Comment (nathanrussell-moj-digital) [inline server/services/educationAndWorkPlanService.ts:69]: Moved this private method to the bottom of the class 👍

**Review by lynnette-orr — APPROVED** [review]

**Pull request merged**